### PR TITLE
Feature Setup Associations

### PR DESCRIPTION
**Pull request by jmonto55**
This PR has the following milestones implemented for the **third** part of the [School Library](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/classes_associations_agregation_composition.md) project:

- Creation of a class Classroom with the following:
    - @label instance variable, should be initialized in the constructor.
    - Setter and getter for @label (remember about attr_accessor).
- Creation of the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
    - Create the has-many side (a classroom has many students).
    - Create the belongs-to side (a student belongs to a classroom).
    - Make sure that when adding a student to a classroom it also sets the classroom for the student.
    - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Creation of a class Book with the following:
    - @title and @author instance variables, should be initialized in the constructor.
    - Setters and getters for instance variables (remember about attr_accessor).
- Creation of a class Rental with the following:
    - @date instance variable, should be initialized in the constructor.
    - Setter and getter for @date (remember about attr_accessor).
- Creation of the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
    - Create the has-many side of Book and Rental (a book has many rentals).
    - Create the belongs-to side of Rental and Book (a rental belongs to a book).
    - Create the has-many side of Person and Rental (a person has many rentals).
    - Create the belongs-to side of Rental and Person (a rental belongs to a person).
    - Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
+
```
Comment (TadesseDev) [inline rental.rb]: - [x] Good job applying the getters and getters @jmonto55. But please remember for this project we need `Getters and Setters for @book and @person instance variables `.  Please add the `Setters` method for @book and @person.

File: person.rb [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
 require_relative './capitalize_decorator'
 require_relative './trimmer_decorator'
+require_relative './rental'
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
 
   def initialize(age, name = 'unknown', parent_permission: true)
```
Comment (TadesseDev) [inline person.rb]: - [x] Also, here, we need to add the `Getters and Setters for @rentals instance variables `.  Please add the `Getters and Setters` method @rentals instance variables.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @jmonto55 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:✔️ ✔️ 

Hi @jmonto55 ✔️ 

Your project is complete! Great job  🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request✔️ 
- Associations added✔️ 
- Methods ok ✔️ 
- No linter errors✔️ 

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**